rbd: update volume/snapshot-volume details on the image as metadata

pkalever opened the pull request:
<!-- Please take a look at our [Contributing](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#Code-contribution-workflow)
documentation before submitting a Pull Request!
Thank you for contributing to ceph-csi! -->

# Describe what this PR does #

 This helps Monitoring solutions without access to Kubernetes clusters to display the details of the PV/PVC/PVCNameSpace and snapshot-name/snapshot-namespace/snapshotcontent-name in their dashboard.

## Is there anything that requires special attention ##

Scenarios covered:
* Validate metadata is present after Create of a PV
* Validate metadata is set after Provisioner pod is restarted
* Validate metadata is updated after a PVC is deleted and PV is reattached to a new PVC
* Validate metadata is updated after a PVC is deleted and PV is reattached to a new PVC on a different namespace
* Validate metadata is set on snapshot create


## Related issues ##
Fixes: #2874

## Other details ##
Currently, this PR uses K8s CO specific keywords
```
// PV metadata keys
pvcNameKey      = "csi.storage.k8s.io/pvc/name"
pcNamespaceKey = "csi.storage.k8s.io/pvc/namespace"
pvNameKey       = "csi.storage.k8s.io/pv/name"

// snapshot metadata keys                                              
volSnapNameKey        = "csi.storage.k8s.io/volumesnapshot/name"       
volSnapNamespaceKey   = "csi.storage.k8s.io/volumesnapshot/namespace"  
volSnapContentNameKey = "csi.storage.k8s.io/volumesnapshotcontent/name"

```
we might want to use generalized naming for keywords, let's discuss it on this PR.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

Commit:
rbd: set PV/PVC details on the image as metadata on create

This helps Monitoring solutions without access to Kubernetes clusters to
display the details of the PV/PVC/NameSpace in their dashboard.

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

Commit:
rbd: update PV/PVC metadata on a reattach of PV

Example if a PVC was delete by setting `persistentVolumeReclaimPolicy` as
`Retain` on PV, and PV is reattached to a new PVC, we make sure to update
PV/PVC image metadata on a PV reattach.

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

Commit:
rbd: set metadata on restart of provisioner pod

Make sure to set metadata when image exist, i.e. if the provisioner pod
is restarted while createVolume is in progress, say it created the image
but didn't yet set the metadata.

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

Commit:
e2e: add test cases for image metadata validation

* create a PVC and check PVC/PV metadata on RBD image
* create and delete a PVC, attach the old PV to a new PVC and check if
  PVC metadata is updated on  RBD image

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

Commit:
rbd: add set/Get VolumeMetadata() utility function

Define and use PV and PVC metadata keys used by external provisioner.
The CSI external-provisioner (v1.6.0+) introduces the
--extra-create-metadata flag, which automatically sets map<string, string>
parameters in the CSI CreateVolumeRequest.

Add utility functions to set/Get PV/PVC/PVCNamespace metadata on image

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

pkalever commented:
/retest all

pkalever commented:
/retest ci/centos/mini-e2e/k8s-1.23

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.23

yati1998 reviewed (commented):
This looks good to me.

Madhu-1 commented on charts/ceph-csi-rbd/templates/provisioner-deployment.yaml:
```diff
@@ -150,6 +150,9 @@ spec:
             {{- end }}
+            {{- if .Values.provisioner.updatemetadata.enabled }}
+            - "--updatemetadata={{ .Values.provisioner.updatemetadata.enabled }}"
+            {{- end }}
```
```suggestion
    - "--setmetadata={{ .Values.provisioner.setmetadata }}"
```

Madhu-1 commented on charts/ceph-csi-rbd/values.yaml:
```diff
@@ -202,6 +202,10 @@ provisioner:
 
+  updatemetadata:
+    # enable updating PC/PVC metadata on RBD image
+    enabled: true
```
just use `setmetadata` to true or false. no need to have enabled flag here?

Madhu-1 commented on cmd/cephcsi.go:
```diff
@@ -65,6 +65,11 @@ func init() {
 	flag.StringVar(&conf.StagingPath, "stagingpath", defaultStagingPath, "staging path")
+	flag.BoolVar(
+		&conf.UpdateMetadata,
+		"updatemetadata",
```
use `setmetadata`? no need to mention PVC/PV/Namespace etc as we can use the same flag for snapshot also.

Madhu-1 commented on cmd/cephcsi.go:
```diff
@@ -65,6 +65,11 @@ func init() {
+		&conf.UpdateMetadata,
+		"updatemetadata",
+		false,
+		"update PV/PVC/PVCNamespace details on the backend image as metadata")
```
`backend image` to `backend volume` as its applicable for both cephfs and rbd

pkalever replied:
I will just make it simple! "set metadata on the volume"

Madhu-1 commented on e2e/rbd.go:
```diff
@@ -394,6 +401,197 @@ var _ = Describe("RBD", func() {
+				}
+
+				pvcName, stdErr, err := execCommandInToolBoxPod(f,
+					fmt.Sprintf("rbd image-meta get %s --image=%s %s",
```
use `rbd image-meta get %s %s/%s` the format should be `pool/rbdimage` as its the format we follow?

pkalever replied:
unfortunately, that was not working hence I had work around this way :-(

Madhu-1 replied:
you just need to change the positions like below

```bash
sh-4.4$ rbd image-meta set replicapool/csi-vol-41f5f225-9f80-11ec-bea5-0242ac110004 test testuibn 
sh-4.4$ rbd image-meta get replicapool/csi-vol-41f5f225-9f80-11ec-bea5-0242ac110004 test
testuibn
```

pkalever replied:
I know, it's failing with that Madhu. Let me prove it for you :-)

Madhu-1 commented on e2e/rbd.go:
```diff
@@ -394,6 +401,197 @@ var _ = Describe("RBD", func() {
+					e2elog.Failf("failed to list rbd images: %v", err)
+				}
+
+				pvcName, stdErr, err := execCommandInToolBoxPod(f,
+					fmt.Sprintf("rbd image-meta get %s --image=%s %s",
+						rbdOptions(defaultRBDPool), imageList[0], pvcNameKey),
+					rookNamespace)
+				if err != nil || stdErr != "" {
+					e2elog.Failf("failed to get PVC name %s/%s %s: err=%v stdErr=%q",
+						rbdOptions(defaultRBDPool), imageList[0], pvcNameKey, err, stdErr)
+				}
+				pvcName = strings.TrimSuffix(pvcName, "\n")
+				if pvcName != pvc.Name {
+					e2elog.Failf("expected pvcName %q got %q", pvc.Name, pvcName)
+				}
```
we dont need to revalidate it's already done the above test?

pkalever replied:
Oh! this is for the test sanity, for better debuggability, nothing wrong in catching it early huh!

Madhu-1 replied:
Yes but its a double check. should be fine if you want to keep it

Madhu-1 commented on e2e/rbd.go:
```diff
@@ -394,6 +401,197 @@ var _ = Describe("RBD", func() {
+				// validate created backend rbd images
+				validateRBDImageCount(f, 1, defaultRBDPool)
+
+				patchBytes = []byte(`{"spec":{"claimRef": null}}`)
+				_, err = c.CoreV1().PersistentVolumes().Patch(
+					context.TODO(),
+					pvcObj.Spec.VolumeName,
+					types.StrategicMergePatchType,
+					patchBytes,
+					metav1.PatchOptions{})
+				if err != nil {
+					e2elog.Logf("error Patching PV %q for claimRef: %v", pvcObj.Spec.VolumeName, err)
+				}
```
we can patch/update only once instead of twice?

Madhu-1 commented on internal/controller/persistentvolume/persistentvolume.go:
```diff
@@ -171,7 +175,25 @@ func (r ReconcilePersistentVolume) reconcilePV(ctx context.Context, obj runtime.
+		claimName = pv.Spec.ClaimRef.Name
+		claimNamespace = pv.Spec.ClaimRef.Namespace
+	} else {
+		claimName = pv.Spec.CSI.VolumeAttributes[pvcNameKey]
+		claimNamespace = pv.Spec.CSI.VolumeAttributes[pvcNamespaceKey]
+	}
```
we should not get pvc name and namespace form volumeattributes. this will be removed as part of https://github.com/ceph/ceph-csi/issues/2835

pkalever replied:
If `claimRef` is `nil` where else can we find them?

Madhu-1 replied:
if claimRef is nil means the PV is not attached to any PVC and we dont need to worry about it as we don't know the actual PVC name and namespace to update the metadata?

Madhu-1 commented on internal/rbd/rbd_util.go:
```diff
@@ -1880,3 +1886,35 @@ func genVolFromVolIDWithMigration(
 }
+
+// Update PV/PVC/PVCNamespace metadata on image.
+func updatePVCandPVMetadata(
```
make this as a method of `rbdVolume`

Madhu-1 commented on internal/rbd/controllerserver.go:
```diff
@@ -315,6 +318,20 @@ func (cs *ControllerServer) CreateVolume(
+	// Update Metadata on create PV
+	if cs.UpdateMetadata {
+		parameters := req.GetParameters()
+		err = updatePVCandPVMetadata(
```
send parameter as input instead of sending separately?

Madhu-1 commented on internal/rbd/rbd_journal.go:
```diff
@@ -530,10 +530,13 @@ func undoVolReservation(ctx context.Context, rbdVol *rbdVolume, cr *util.Credent
 // Generate new volume Handler
 // The volume handler won't remain same as its contains poolID,clusterID etc
 // which are not same across clusters.
+// nolint:gocyclo,nestif,cyclop // reduce complexity
```
avoid adding this one? try refractoring

pkalever replied:
The diff is too small and I doubt we will have any benefit by moving it into a new function. Lets live with it for now as refactoring the entire function is out of scope to this PR

Madhu-1 replied:
Yes, agree. as I mentioned earlier sadly it's not recomended to add nolint , max we try to avoid adding it because if we do so no limit for adding more complexity

Madhu-1 commented on internal/util/util.go:
```diff
@@ -68,57 +68,53 @@ var (
 	// on rbd image without flattening, once the limit is reached cephcsi will
 	// start flattening the older rbd images to allow more snapshots
 	MaxSnapshotsOnImage uint
-
```
not part of this change can this be reverted back?

pkalever replied:
Nope, the linter will be upset. Check the commit-msg if you have missed it.

Madhu-1 replied:
I did same changes, it's not failing for me. :thinking: 

```go
[🎩︎]mrajanna@fedora ceph-csi $]make go-lint
cephcsi image settings: quay.io/cephcsi/cephcsi version canary
./scripts/lint-go.sh
INFO [config_reader] Used config file scripts/golangci.yml 
INFO [lintersdb] Active 63 linters: [asciicheck bidichk bodyclose cyclop deadcode depguard dogsled durationcheck errcheck errname errorlint exhaustive exportloopref forcetypeassert gocognit goconst gocritic gocyclo godot gofmt gofumpt goheader goimports golint gomodguard goprintffuncname gosec gosimple govet ifshort importas ineffassign interfacer lll makezero maligned misspell nakedret nestif nilerr nlreturn noctx nolintlint paralleltest predeclared promlinter revive rowserrcheck scopelint sqlclosecheck staticcheck structcheck stylecheck tenv thelper tparallel typecheck unconvert unparam unused varcheck wastedassign whitespace] 
INFO [loader] Using build tags: [pacific]         
INFO [loader] Go packages loading at mode 575 (compiled_files|deps|exports_file|imports|name|types_sizes|files) took 3.543044639s 
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref. 
WARN [runner] The linter 'maligned' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  Replaced by govet 'fieldalignment'. 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 42.586643ms 
INFO [linters context] importas settings found, but no aliases listed. List aliases under alias: key. 
INFO [linters context/goanalysis] analyzers took 1m41.518813007s with top 10 stages: gocritic: 30.226675639s, buildir: 7.597464563s, the_only_name: 7.219152421s, wastedassign: 5.488997292s, buildssa: 5.395790241s, gofumpt: 3.694478959s, goimports: 3.33554207s, gofmt: 2.087923004s, golint: 2.022580381s, gosec: 2.004654201s 
INFO [runner] Issues before processing: 513, after processing: 0 
INFO [runner] Processors filtering stat (out/in): cgo: 513/513, path_prettifier: 513/513, autogenerated_exclude: 513/513, identifier_marker: 513/513, filename_unadjuster: 513/513, skip_dirs: 513/513, exclude: 513/513, exclude-rules: 181/513, nolint: 0/181, skip_files: 513/513 
INFO [runner] processing took 85.550267ms with stages: nolint: 46.458945ms, exclude-rules: 12.957937ms, identifier_marker: 12.678271ms, autogenerated_exclude: 11.027662ms, path_prettifier: 2.031872ms, skip_dirs: 290.513µs, filename_unadjuster: 50.715µs, cgo: 47.344µs, max_same_issues: 2.278µs, skip_files: 801ns, uniq_by_line: 638ns, severity-rules: 614ns, max_from_linter: 487ns, diff: 411ns, source_code: 379ns, max_per_file_from_linter: 365ns, exclude: 335ns, sort_results: 259ns, path_shortener: 241ns, path_prefixer: 200ns 
INFO [runner] linters took 13.373613913s with stages: goanalysis_metalinter: 13.287932562s 
INFO File cache stats: 116 entries of total size 1.0MiB 
INFO Memory: 157 samples, avg is 779.2MB, max is 1382.8MB 
INFO Execution took 16.972425146s                 
[🎩︎]mrajanna@fedora ceph-csi $]git diff
diff --git a/cmd/cephcsi.go b/cmd/cephcsi.go
index f419790b0..d8b200538 100644
--- a/cmd/cephcsi.go
+++ b/cmd/cephcsi.go
@@ -117,6 +117,11 @@ func init() {
                "maxsnapshotsonimage",
                450,
                "Maximum number of snapshots allowed on rbd image without flattening")
+       flag.BoolVar(
+               &conf.SetMetadata,
+               "setmetadata",
+               false,
+               "set metadata on the volume")
        flag.UintVar(
                &conf.MinSnapshotsOnImage,
                "minsnapshotsonimage",
diff --git a/internal/util/util.go b/internal/util/util.go
index c67fd0570..2995be051 100644
--- a/internal/util/util.go
+++ b/internal/util/util.go
@@ -101,6 +101,8 @@ type Config struct {
        // cephfs related flags
        ForceKernelCephFS bool // force to use the ceph kernel client even if the kernel is < 4.17
 
+       SetMetadata bool // set metadata on the volume
+
        // RbdHardMaxCloneDepth is the hard limit for maximum number of nested volume clones that are taken before a flatten
        // occurs
        RbdHardMaxCloneDepth uint
[🎩︎]mrajanna@fedora ceph-csi $]
```

pkalever replied:
I don't know if that is related to the linter version that CI is using, but the CI logs had it and I had captured some part of it the commit msg.

Commit:
rbd: set metadata on the snapshot

Set snapshot-name/snapshot-namespace/snapshotcontent-name details
on RBD backend snapshot image as metadata on snapshot

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

Commit:
e2e: add test case for snapshot metadata validation

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

pkalever reviewed (commented):
Thanks Madhu

Commit:
deploy: add --extra-create-metadata arg to csi-snapshotter sidecar

This argument in csi-snapshotter sidecar allows us to receive
snapshot-name/snapshot-namespace/snapshotcontent-name metadata in the
CreateSnapshot() request.

For ex:

csi.storage.k8s.io/volumesnapshot/name
csi.storage.k8s.io/volumesnapshot/namespace
csi.storage.k8s.io/volumesnapshotcontent/name

This is a useful information which can be used depend on the use case we
have at our driver. The features like adding metadata to snapshot image
can consume this based on the need.

Signed-off-by: Prasanna Kumar Kalever <prasanna.kalever@redhat.com>

pkalever commented:
/retest ci/centos/mini-e2e/k8s-1.22

Madhu-1 commented on internal/rbd/controllerserver.go:
```diff
@@ -342,8 +353,13 @@ func flattenParentImage(ctx context.Context, rbdVol *rbdVolume, cr *util.Credent
+	cr *util.Credentials,
+	rbdVol *rbdVolume,
+	rbdSnap *rbdSnapshot,
+	setMetadata bool) (*csi.CreateVolumeResponse, error) {
```
you dont need to pass metMetadata use cs.SetMetadata from the receiver? instead of request just send parameters as input to this function?

Madhu-1 commented on internal/rbd/controllerserver.go:
```diff
@@ -378,6 +394,14 @@ func (cs *ControllerServer) repairExistingVolume(ctx context.Context, req *csi.C
 			return nil, err
 		}
+
+		// Set metadata on restart of provisioner pod when image exist
+		if setMetadata {
+			err = rbdVol.setImageMetadata(ctx, req.GetParameters())
+			if err != nil {
+				return nil, err
+			}
+		}
```
setmetadata should be done outside as it's applicable for both volume clone and volume creation

Madhu-1 commented on internal/rbd/controllerserver.go:
```diff
@@ -968,8 +993,9 @@ func (cs *ControllerServer) CreateSnapshot(
 
 		return nil, status.Errorf(codes.Internal, err.Error())
 	}
+	parameters := req.GetParameters()
```
snapshotcontent can be detatched and reattached to a new snapshot in a different namespace. do we have code to regenerate the metadata in that case?

pkalever replied:
I have not checked this part, AI on me

Madhu-1 replied:
lets open up a new issue to track it. we need new controller for this one. we can do a follow up

Madhu-1 commented on internal/rbd/controllerserver.go:
```diff
@@ -896,6 +920,7 @@ func (cs *ControllerServer) ValidateVolumeCapabilities(
 }
 
 // CreateSnapshot creates the snapshot in backend and stores metadata in store.
+// nolint:gocyclo,cyclop // reduce complexity
```
add nolint is not encouraged as it gets staled once this PR is merged. common practice is to do refactor and add the functionality.

Madhu-1 commented on internal/rbd/controllerserver.go:
```diff
@@ -995,6 +1021,17 @@ func (cs *ControllerServer) CreateSnapshot(
+		rbdVol.RbdImageName = rbdSnap.RbdSnapName
+		err = rbdVol.setSnapShotMetadata(ctx, req.GetParameters())
+		if err != nil {
+			return nil, err
```
return GRPC error message

Madhu-1 commented on internal/rbd/controllerserver.go:
```diff
@@ -1045,6 +1084,15 @@ func cloneFromSnapshot(
+	if updateMetadata {
+		err = rbdVol.setSnapShotMetadata(ctx, parameters)
+		if err != nil {
+			return nil, err
```
return GRPC error message?

Madhu-1 commented on internal/rbd/rbd_journal.go:
```diff
@@ -585,7 +588,7 @@ func RegenerateJournal(
 	}
 
-	rbdVol.RequestName = requestName
+	rbdVol.RequestName = volumeAttributes[pvNameKey]
```
pv name will be removed from the volumeAttributers in sometime as i mentioned ealier

Madhu-1 commented on internal/rbd/rbd_journal.go:
```diff
@@ -610,6 +614,19 @@ func RegenerateJournal(
+		// Update Metadata on reattach of the same old PV
+		if setMetadata &&
+			(claimName != volumeAttributes[pvcNameKey] || claimNamespace != volumeAttributes[pvcNamespaceKey]) {
+			parameters := map[string]string{
+				pvcNameKey:      claimName,
+				pvcNamespaceKey: claimNamespace,
+			}
+			err = rbdVol.setImageMetadata(ctx, parameters)
+			if err != nil {
+				return "", err
+			}
```
move this logic to line 693 after storing the imageID?

Madhu-1 commented on internal/rbd/rbd_util.go:
```diff
@@ -1882,3 +1893,40 @@ func genVolFromVolIDWithMigration(
 }
+
+// Set PV/PVC/PVCNamespace metadata on RBD image.
+func (rv *rbdVolume) setImageMetadata(
```
you might need to rename the funtion as this is specific to setting PVC-related information?

pkalever replied:
renaming it as setVolumeMetadata()

pkalever reviewed (commented):
Resolved most of the comments. Thanks!

Madhu-1 commented on internal/rbd/controllerserver.go:
```diff
@@ -995,6 +1015,17 @@ func (cs *ControllerServer) CreateSnapshot(
+	if cs.SetMetadata {
+		// Update the metadata on snapshot not on the original image
+		rbdVol.RbdImageName = rbdSnap.RbdSnapName
+		err = rbdVol.setSnapShotMetadata(ctx, req.GetParameters())
```
During rbd clone operation all the clones will get the metadata from the parent we need to unset/remove the metadata in below cases

* During CreateSnapshot ->Unset the PVC metadata on the rbd image created for the snapshot
* During CreateVolume From snapshot -> Unset the snapshot metadata from the rbd image created from the snapshot
* During CreateVolume From Volume -> Unset the parent PVC metadata on the temp clone rbd image. 

If this looks too much open up an issue to track we can fix this in followup PR's.

pkalever commented:
@Madhu-1 previously the reattach case was working fine and e2e was passing, after the rebase its failing any lead please?

Madhu-1 commented:
> @Madhu-1 previously the reattach case was working fine and e2e was passing, after the rebase its failing any lead please?

```
Mar 29 10:40:26.687: INFO: PersistentVolume pvc-a9a107d4-447f-41d7-81d3-4cdade20a21d found and phase=Bound (1.618557ms)
Mar 29 10:40:26.694: INFO: ExecWithOptions {Command:[/bin/sh -c rbd ls --format=json --pool=replicapool] Namespace:rook-ceph PodName:rook-ceph-tools-696bff5bc6-v4hd6 ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Mar 29 10:40:26.694: INFO: >>> kubeConfig: /root/.kube/config
Mar 29 10:40:26.695: INFO: ExecWithOptions: Clientset creation
Mar 29 10:40:26.695: INFO: ExecWithOptions: execute(POST https://192.168.39.115:8443/api/v1/namespaces/rook-ceph/pods/rook-ceph-tools-696bff5bc6-v4hd6/exec?command=%2Fbin%2Fsh&command=-c&command=rbd+ls+--format%3Djson+--pool%3Dreplicapool&container=rook-ceph-tools&container=rook-ceph-tools&stderr=true&stdout=true %!s(MISSING))
Mar 29 10:40:27.073: INFO: ExecWithOptions {Command:[/bin/sh -c rbd ls --format=json --pool=replicapool] Namespace:rook-ceph PodName:rook-ceph-tools-696bff5bc6-v4hd6 ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Mar 29 10:40:27.073: INFO: >>> kubeConfig: /root/.kube/config
Mar 29 10:40:27.074: INFO: ExecWithOptions: Clientset creation
Mar 29 10:40:27.074: INFO: ExecWithOptions: execute(POST https://192.168.39.115:8443/api/v1/namespaces/rook-ceph/pods/rook-ceph-tools-696bff5bc6-v4hd6/exec?command=%2Fbin%2Fsh&command=-c&command=rbd+ls+--format%3Djson+--pool%3Dreplicapool&container=rook-ceph-tools&container=rook-ceph-tools&stderr=true&stdout=true %!s(MISSING))
Mar 29 10:40:27.447: INFO: ExecWithOptions {Command:[/bin/sh -c rbd image-meta get --pool=replicapool --image=csi-vol-4254aac3-af44-11ec-a96f-5a9681a400cb csi.storage.k8s.io/pvc/name] Namespace:rook-ceph PodName:rook-ceph-tools-696bff5bc6-v4hd6 ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Mar 29 10:40:27.447: INFO: >>> kubeConfig: /root/.kube/config
Mar 29 10:40:27.448: INFO: ExecWithOptions: Clientset creation
Mar 29 10:40:27.448: INFO: ExecWithOptions: execute(POST https://192.168.39.115:8443/api/v1/namespaces/rook-ceph/pods/rook-ceph-tools-696bff5bc6-v4hd6/exec?command=%2Fbin%2Fsh&command=-c&command=rbd+image-meta+get+--pool%3Dreplicapool+--image%3Dcsi-vol-4254aac3-af44-11ec-a96f-5a9681a400cb+csi.storage.k8s.io%2Fpvc%2Fname&container=rook-ceph-tools&container=rook-ceph-tools&stderr=true&stdout=true %!s(MISSING))
Mar 29 10:40:27.868: INFO: ExecWithOptions {Command:[/bin/sh -c rbd ls --format=json --pool=replicapool] Namespace:rook-ceph PodName:rook-ceph-tools-696bff5bc6-v4hd6 ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Mar 29 10:40:27.868: INFO: >>> kubeConfig: /root/.kube/config
Mar 29 10:40:27.868: INFO: ExecWithOptions: Clientset creation
Mar 29 10:40:27.868: INFO: ExecWithOptions: execute(POST https://192.168.39.115:8443/api/v1/namespaces/rook-ceph/pods/rook-ceph-tools-696bff5bc6-v4hd6/exec?command=%2Fbin%2Fsh&command=-c&command=rbd+ls+--format%3Djson+--pool%3Dreplicapool&container=rook-ceph-tools&container=rook-ceph-tools&stderr=true&stdout=true %!s(MISSING))
Mar 29 10:40:28.259: INFO: Waiting up to &PersistentVolumeClaim{ObjectMeta:{rbd-pvc-new  rbd-1318    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Spec:PersistentVolumeClaimSpec{AccessModes:[ReadWriteOnce],Resources:ResourceRequirements{Limits:ResourceList{},Requests:ResourceList{storage: {{1073741824 0} {<nil>} 1Gi BinarySI},},},VolumeName:,Selector:nil,StorageClassName:*csi-rbd-sc,VolumeMode:nil,DataSource:nil,DataSourceRef:nil,},Status:PersistentVolumeClaimStatus{Phase:,AccessModes:[],Capacity:ResourceList{},Conditions:[]PersistentVolumeClaimCondition{},AllocatedResources:ResourceList{},ResizeStatus:nil,},} to be in Bound state
Mar 29 10:40:28.259: INFO: waiting for PVC rbd-pvc-new (0 seconds elapsed)
Mar 29 10:40:30.261: INFO: waiting for PVC rbd-pvc-new (2 seconds elapsed)
Mar 29 10:40:30.266: INFO: Waiting for PV pvc-cf6ab5d2-9fd9-4cd6-ae16-e89409656174 to bind to PVC rbd-pvc-new
```

Looks like a new PV is getting created when you delete and recreate the PVC.

Madhu-1 commented on e2e/rbd.go:
```diff
@@ -394,6 +406,296 @@ var _ = Describe("RBD", func() {
+				// validate created backend rbd images
+				validateRBDImageCount(f, 1, defaultRBDPool)
+
+				pvc.Name = "rbd-pvc-new"
```
use `pvcObj` instead of pvc here.

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.23

Rakshith-R commented:
Pass `false` or comment the line in values.yaml and deployment files ? 
Let users enable it, if they need it.

Rakshith-R commented on internal/rbd/rbd_journal.go:
```diff
@@ -598,16 +604,23 @@ func RegenerateJournal(
 		rbdVol.ReservedID = imageData.ImageUUID
 		rbdVol.ImageID = imageData.ImageAttributes.ImageID
 		rbdVol.Owner = imageData.ImageAttributes.Owner
+		rbdVol.RbdImageName = imageData.ImageAttributes.ImageName
```
I dont think we need this line

pkalever replied:
SetMetadata() needs it https://github.com/ceph/ceph-csi/blob/devel/internal/rbd/rbd_util.go#L1663

pkalever commented:
> Pass `false` or comment the line in values.yaml and deployment files ? Let users enable it, if they need it.

These are used by our e2e right? I want the e2e to test the same, probably `ceph-csi/example/` is intended for users

Rakshith-R commented on internal/rbd/rbd_journal.go:
```diff
@@ -533,10 +533,16 @@ func undoVolReservation(ctx context.Context, rbdVol *rbdVolume, cr *util.Credent
-	volumeID, requestName, owner string,
-	cr *util.Credentials) (string, error) {
+	claimName,
+	claimNamespace,
+	volumeID,
+	requestName,
+	owner string,
```
```suggestion
	volumeID,
	requestName,
	owner string,
```
owner refers to the namespace.

nixpanic commented on internal/rbd/rbd_journal.go:
```diff
@@ -652,6 +665,28 @@ func RegenerateJournal(
 	return rbdVol.VolID, nil
 }
 
+// set volume details as Image metadata.
```
comments for functions usually start with the function name

nixpanic commented on internal/rbd/rbd_util.go:
```diff
@@ -1918,3 +1929,40 @@ func genVolFromVolIDWithMigration(
+func (rv *rbdVolume) setVolumeMetadata(
+	ctx context.Context, parameters map[string]string) error {
+	s := []string{pvcNameKey, pvcNamespaceKey, pvNameKey}
+	for _, i := range s {
```
`s` is a bad name for a variable, maybe use `keys`?
`i` is usually a counter, maybe use `key` instead?

nixpanic commented on internal/rbd/rbd_util.go:
```diff
@@ -1918,3 +1929,40 @@ func genVolFromVolIDWithMigration(
+
+// Set snapshot-name/snapshot-namespace/snapshotcontent-name metadata
+// on RBD image.
+func (rv *rbdVolume) setSnapShotMetadata(
```
`Snapshot` is commonly used, not `SnapShot`.

nixpanic commented on internal/rbd/rbd_util.go:
```diff
@@ -1918,3 +1929,40 @@ func genVolFromVolIDWithMigration(
+		if parameters[i] != "" {
+			err := rv.SetMetadata(i, parameters[i])
+			if err != nil {
+				log.ErrorLog(ctx, "failed to set metadata key %q, value %q on image: %v", i, parameters[i], err)
```
instead of logging, just return the error with the additional context?

nixpanic commented on internal/rbd/rbd_journal.go:
```diff
@@ -652,6 +665,28 @@ func RegenerateJournal(
+		}
+		err := rv.setVolumeMetadata(ctx, parameters)
+		if err != nil {
+			return err
```
ideally add some context to the error that you return

nixpanic reviewed (commented):
just some nits, nothing that is blocking from my point of view

nixpanic commented:
@pkalever please review your commit messages for typos (documentation in subject), and be more verbose for the "add testcases" one.

nixpanic commented on cmd/cephcsi.go:
```diff
@@ -68,6 +68,11 @@ func init() {
 	flag.StringVar(&conf.NodeID, "nodeid", "", "node id")
 	flag.StringVar(&conf.PluginPath, "pluginpath", defaultPluginPath, "plugin path")
 	flag.StringVar(&conf.StagingPath, "stagingpath", defaultStagingPath, "staging path")
+	flag.BoolVar(
```
Just wondering, but have you considered to just set the metadata in case it is available?

pkalever replied:
Can you explain a bit more, I didn't get what you mean by `set metadata in case it is available`?

pkalever replied:
@nixpanic ?

Madhu-1 replied:
what Niels meant was to create the metadata if it's available in the request. let's not have a flag for this one as it will be useful to have the metadata on the ceph resources. is that correct @nixpanic? or set the metadata if it's available and the flag is set to true?

nixpanic commented on internal/rbd/controllerserver.go:
```diff
@@ -1086,7 +1099,9 @@ func cloneFromSnapshot(
-	cr *util.Credentials) (*csi.CreateSnapshotResponse, error) {
+	cr *util.Credentials,
+	parameters map[string]string,
+	updateMetadata bool) (*csi.CreateSnapshotResponse, error) {
```
instead of passing the additional bool, just use `nil` or an empty map as `parameters`?

Madhu-1 commented on e2e/rbd.go:
```diff
@@ -394,6 +406,298 @@ var _ = Describe("RBD", func() {
+					}
+				}()
+
+				// create PVC and bind it to an app
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					e2elog.Failf("failed to load PVC: %v", err)
+				}
+
+				pvc.Namespace = f.UniqueName
+				app, err := loadApp(appPath)
+				if err != nil {
+					e2elog.Failf("failed to load application: %v", err)
+				}
+				app.Namespace = f.UniqueName
+				err = createPVCAndApp("", f, pvc, app, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create PVC and application: %v", err)
+				}
```
use `createPVCAndAppBinding` helper function

Madhu-1 commented on internal/rbd/rbd_journal.go:
```diff
@@ -598,16 +603,23 @@ func RegenerateJournal(
 	if err != nil {
 		return "", err
 	}
+
 	if imageData != nil {
 		rbdVol.ReservedID = imageData.ImageUUID
 		rbdVol.ImageID = imageData.ImageAttributes.ImageID
 		rbdVol.Owner = imageData.ImageAttributes.Owner
+		rbdVol.RbdImageName = imageData.ImageAttributes.ImageName
 		if rbdVol.ImageID == "" {
 			err = rbdVol.storeImageID(ctx, j)
 			if err != nil {
 				return "", err
 			}
 		}
+		// Update Metadata on reattach of the same old PV
+		err = rbdVol.setVolumeDetailsAsMetadata(volumeAttributes, claimName, rbdVol.Owner, setMetadata)
```
owner is already member of rbdVol. we don't need to sent it as input to `setVolumeDetailsAsMetadata`

Madhu-1 replied:
call this only if setMetadata is set here only we don't need to pass that as a argument?

Madhu-1 commented on internal/rbd/rbd_journal.go:
```diff
@@ -652,6 +664,27 @@ func RegenerateJournal(
+	claimNamespace string,
+	setMetadata bool) error {
+	if setMetadata &&
+		claimName != volumeAttributes[pvcNameKey] || claimNamespace != volumeAttributes[pvcNamespaceKey] {
```
This check is not required as we don't store the PVC name and namespace name in the volume attributes, I think you also need to set the PV name also as once can create a  new PV for the existing rbd image and use it for the new PVC.

pkalever replied:
We don't store PVC name and namespace name in the volume attributes, with your recent changes?

Madhu-1 replied:
Yes correct

Madhu-1 reviewed (commented):
some nits, LGTM

nixpanic commented:
On Fri, Apr 01, 2022 at 04:44:56AM -0700, Madhu Rajanna wrote:
> @Madhu-1 commented on this pull request.
> 
> 
> 
> > @@ -68,6 +68,11 @@ func init() {
>  	flag.StringVar(&conf.NodeID, "nodeid", "", "node id")
>  	flag.StringVar(&conf.PluginPath, "pluginpath", defaultPluginPath, "plugin path")
>  	flag.StringVar(&conf.StagingPath, "stagingpath", defaultStagingPath, "staging path")
> +	flag.BoolVar(
> 
> what Niels meant was to create the metadata if it's available in the request. let's not have a flag for this one as it will be useful to have the metadata on the ceph resources. is that correct @nixpanic? or set the metadata if it's available and the flag is set to true?

Indeed, I do not think there is a need for this flag. If the metadata is
passed to the CreateVolume call, just set it. Not all options need to be
configurable, that makes it just more complicated for users.

pkalever commented:
> On Fri, Apr 01, 2022 at 04:44:56AM -0700, Madhu Rajanna wrote: @Madhu-1 commented on this pull request. > @@ -68,6 +68,11 @@ func init() { flag.StringVar(&conf.NodeID, "nodeid", "", "node id") flag.StringVar(&conf.PluginPath, "pluginpath", defaultPluginPath, "plugin path") flag.StringVar(&conf.StagingPath, "stagingpath", defaultStagingPath, "staging path") + flag.BoolVar( what Niels meant was to create the metadata if it's available in the request. let's not have a flag for this one as it will be useful to have the metadata on the ceph resources. is that correct @nixpanic? or set the metadata if it's available and the flag is set to true?
> Indeed, I do not think there is a need for this flag. If the metadata is passed to the CreateVolume call, just set it. Not all options need to be configurable, that makes it just more complicated for users.

Which means we are depending on `--extra-create-metadata=true` option of the external provisioner, this option will dump other details too, what if the user is interested in other details? we might unnecessarily add metadata to images wasting a few jiffies of performance.

Madhu-1 commented:
> > On Fri, Apr 01, 2022 at 04:44:56AM -0700, Madhu Rajanna wrote: @Madhu-1 commented on this pull request. > @@ -68,6 +68,11 @@ func init() { flag.StringVar(&conf.NodeID, "nodeid", "", "node id") flag.StringVar(&conf.PluginPath, "pluginpath", defaultPluginPath, "plugin path") flag.StringVar(&conf.StagingPath, "stagingpath", defaultStagingPath, "staging path") + flag.BoolVar( what Niels meant was to create the metadata if it's available in the request. let's not have a flag for this one as it will be useful to have the metadata on the ceph resources. is that correct @nixpanic? or set the metadata if it's available and the flag is set to true?
> > Indeed, I do not think there is a need for this flag. If the metadata is passed to the CreateVolume call, just set it. Not all options need to be configurable, that makes it just more complicated for users.
> 
> Which means we are depending on `--extra-create-metadata=true` option of the external provisioner, this option will dump other details too, what if the user is interested in other details? we might unnecessarily add metadata to images wasting a few jiffies of performance.

I think we should make adding metadata default as it helps in monitoring, if someone case about performance I think they will also worry about the monitoring. I think there won't be much performance affect by this as we use go-ceph to add the metadata. WDYT?

nixpanic commented:
> we might unnecessarily add metadata to images wasting a few jiffies of performance.

I do not think this will be noticeable. My preference is an as simple as possible solution, rather reducing extra configuration options, and not add more for little benefit.

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

nixpanic commented on internal/rbd/rbd_util.go:
```diff
@@ -78,6 +78,12 @@ const (
+	// create requests as parameters, when `extra-create-metadata` is true.
+	pvcNameKey      = "csi.storage.k8s.io/pvc/name"
+	pvcNamespaceKey = "csi.storage.k8s.io/pvc/namespace"
+	pvNameKey       = "csi.storage.k8s.io/pv/name"
```
These constants should probably move to `internal/util/k8s/parameters.go`. Add a function
```go
func GetVolumeMetadata(parameters map[string]string) map[string]string {
    // filter parameters, only return volume metadata
}
```
there too

nixpanic commented on internal/rbd/rbd_util.go:
```diff
@@ -1918,3 +1924,18 @@ func genVolFromVolIDWithMigration(
+
+// setVolumeMetadata set PV/PVC/PVCNamespace metadata on RBD image.
+func (rv *rbdVolume) setVolumeMetadata(parameters map[string]string) error {
+	keys := []string{pvcNameKey, pvcNamespaceKey, pvNameKey}
```
Use 
```go
for k, v := range k8s.GetVolumeMetadata(parameters) {
    err := rv.SetMetadata(k, v)
}
```

That way the kubernetes specific metadata filtering is kept inside the k8s package.

nixpanic commented on internal/rbd/rbd_journal.go:
```diff
@@ -598,16 +602,27 @@ func RegenerateJournal(
 		}
+		// Update Metadata on reattach of the same old PV
+		parameters := map[string]string{
+			pvcNameKey:      claimName,
```
find a way to use a helper function under the `internal/utils/k8s` package to create the `parameters` map

nixpanic reviewed (changes requested):
some minimal changes in code-organization would be good to have, making things cleaner

nixpanic reviewed (approved):
thanks for the cleanup, looks very good now :+1:

pkalever commented:
@Mergifyio rebase

humblec commented:
/retest ci/centos/mini-e2e/k8s-1.23

humblec commented:
/retest ci/centos/mini-e2e/k8s-1.21

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

nixpanic commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

nixpanic commented:
> /retest ci/centos/mini-e2e-helm/k8s-1.21

Failed with:
```
Apr  4 16:28:50.446: FAIL: backend images not matching kubernetes resource count,image count 23 kubernetes resource count 21
```

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

pkalever commented:
/retest ci/centos/mini-e2e/k8s-1.21

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

humblec commented:
@Mergifyio refresh

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

Madhu-1 commented:
@mergifyio refresh

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

Rakshith-R commented:
@Mergifyio refresh

humblec commented:
@Mergifyio refresh

humblec commented:
@pkalever the tests are not passing.. and its blocking the release.. can you take a look at it.?

Madhu-1 commented:
/retes ci/centos/mini-e2e-helm/k8s-1.21

Madhu-1 commented:
> /retes ci/centos/mini-e2e-helm/k8s-1.21

Failure is due to known reason, @pkalever please check and confirm
```
Apr  5 07:26:20.106: INFO: ExecWithOptions: execute(POST https://192.168.39.36:8443/api/v1/namespaces/rook-ceph/pods/rook-ceph-tools-6f56cdd85d-99zjs/exec?command=%2Fbin%2Fsh&command=-c&command=rbd+ls+--format%3Djson+--pool%3Dreplicapool&container=rook-ceph-tools&container=rook-ceph-tools&stderr=true&stdout=true %!s(MISSING))
Apr  5 07:26:21.140: FAIL: backend images not matching kubernetes resource count,image count 23 kubernetes resource count 21
backend image Info:
 [csi-vol-28727ce8-b4a9-11ec-b5bc-1eb41208b084 csi-vol-3a6e6c62-b4a9-11ec-b5bc-1eb41208b084 csi-vol-3a6e6c62-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-3a899665-b4a9-11ec-b5bc-1eb41208b084 csi-vol-3a899665-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-3aa9b48a-b4a9-11ec-b5bc-1eb41208b084 csi-vol-3aa9b48a-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-3ac8b719-b4a9-11ec-b5bc-1eb41208b084 csi-vol-3ac8b719-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-3aeb2abc-b4a9-11ec-b5bc-1eb41208b084 csi-vol-3aeb2abc-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-3d24312b-b4a9-11ec-b5bc-1eb41208b084 csi-vol-3d24312b-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-3f502b26-b4a9-11ec-b5bc-1eb41208b084 csi-vol-3f502b26-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-3fe67f84-b4a9-11ec-b5bc-1eb41208b084 csi-vol-3fe67f84-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-409d1131-b4a9-11ec-b5bc-1eb41208b084 csi-vol-409d1131-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-40b7db93-b4a9-11ec-b5bc-1eb41208b084 csi-vol-40b7db93-b4a9-11ec-b5bc-1eb41208b084-temp csi-vol-40e2e952-b4a9-11ec-b5bc-1eb41208b084 csi-vol-40e2e952-b4a9-11ec-b5bc-1eb41208b084-temp]
```

Madhu-1 commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

pkalever commented:
@Madhu-1  sorry I didn't get what is known reason? is this PR failing CI for a valid reason? meaning is this PR introducing any regression?

Madhu-1 commented:
> @Madhu-1 sorry I didn't get what is known reason? is this PR failing CI for a valid reason? meaning is this PR introducing any regression?

https://github.com/ceph/ceph-csi/issues/2327 covers the known reason

humblec commented:
hmmm.. Failed again... @pkalever  @Madhu-1

humblec commented:
@Madhu-1 manual merge ?

Madhu-1 commented:
> @Madhu-1 manual merge ?

If we manually merge it will keep happening on all the PRs and it's bad for CI?

Madhu-1 commented:
> hmmm.. Failed again... @pkalever @Madhu-1

This times its failed for something else

```
pr  5 08:45:18.682: INFO: Waiting up to &PersistentVolumeClaim{ObjectMeta:{rbd-pvc-new  rbd-3274  d8d99c66-37a0-4142-85f9-6177bc0cff07  0 2022-04-05 08:45:12 +0100 BST <nil> <nil> map[] map[pv.kubernetes.io/bind-completed:yes pv.kubernetes.io/bound-by-controller:yes volume.beta.kubernetes.io/storage-provisioner:rbd.csi.ceph.com] [] [kubernetes.io/pvc-protection]  [{e2e.test Update v1 2022-04-05 08:45:12 +0100 BST FieldsV1 {"f:spec":{"f:accessModes":{},"f:resources":{"f:requests":{".":{},"f:storage":{}}},"f:storageClassName":{},"f:volumeMode":{}}} } {kube-controller-manager Update v1 2022-04-05 08:45:12 +0100 BST FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:pv.kubernetes.io/bind-completed":{},"f:pv.kubernetes.io/bound-by-controller":{},"f:volume.beta.kubernetes.io/storage-provisioner":{}}},"f:spec":{"f:volumeName":{}},"f:status":{"f:accessModes":{},"f:capacity":{".":{},"f:storage":{}},"f:phase":{}}} }]},Spec:PersistentVolumeClaimSpec{AccessModes:[ReadWriteOnce],Resources:ResourceRequirements{Limits:ResourceList{},Requests:ResourceList{storage: {{1073741824 0} {<nil>} 1Gi BinarySI},},},VolumeName:pvc-d8d99c66-37a0-4142-85f9-6177bc0cff07,Selector:nil,StorageClassName:*csi-rbd-sc,VolumeMode:*Filesystem,DataSource:nil,DataSourceRef:nil,},Status:PersistentVolumeClaimStatus{Phase:Bound,AccessModes:[ReadWriteOnce],Capacity:ResourceList{storage: {{1073741824 0} {<nil>} 1Gi BinarySI},},Conditions:[]PersistentVolumeClaimCondition{},AllocatedResources:ResourceList{},ResizeStatus:nil,},} to be in Bound state
Apr  5 08:45:18.682: INFO: waiting for PVC rbd-pvc-new (0 seconds elapsed)
Apr  5 08:45:18.688: INFO: Waiting for PV pvc-d8d99c66-37a0-4142-85f9-6177bc0cff07 to bind to PVC rbd-pvc-new
Apr  5 08:45:18.688: INFO: Waiting up to timeout=10m0s for PersistentVolumeClaims [rbd-pvc-new] to have phase Bound
Apr  5 08:45:18.691: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Pending instead of Bound.
Apr  5 08:45:20.695: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:22.699: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:24.704: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:26.708: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:28.711: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:30.714: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:32.719: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:34.724: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:36.727: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:38.731: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:40.735: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:42.740: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:44.744: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:46.749: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:48.753: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:50.757: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:52.761: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:54.766: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:56.770: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:45:58.774: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:00.778: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:02.782: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:04.786: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:06.790: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:08.795: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:10.799: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:12.804: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:14.809: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:16.812: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:18.817: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:20.825: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:22.829: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:24.834: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:26.838: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:28.842: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:30.846: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:32.850: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:34.854: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:36.858: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:38.862: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:40.866: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:42.870: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:44.875: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:46.879: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:48.885: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:50.889: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:52.894: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:54.898: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:56.902: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:46:58.907: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:00.911: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:02.916: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:04.921: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:06.926: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:08.930: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:10.934: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:12.939: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:14.946: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:16.950: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:18.953: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:20.957: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:22.963: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:24.967: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:26.972: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:28.977: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:30.981: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:32.985: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:34.990: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:36.995: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:39.001: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:41.006: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:43.010: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:45.015: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:47.021: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:49.025: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:51.030: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:53.034: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:55.039: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:57.044: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:47:59.049: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:01.055: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:03.059: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:05.064: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:07.069: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:09.074: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:11.078: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:13.083: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:15.088: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:17.093: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:19.097: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:21.102: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:23.106: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:25.111: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:27.117: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:29.122: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:31.127: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:33.131: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:35.136: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:37.141: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:39.145: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:41.150: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:43.154: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:45.158: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:47.168: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:49.172: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:51.176: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:53.180: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:55.184: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:57.193: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:48:59.198: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:01.207: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:03.211: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:05.215: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:07.220: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:09.224: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:11.228: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:13.232: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:15.239: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:17.243: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:19.247: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:21.252: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:23.256: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:25.261: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:27.267: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:29.271: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:31.276: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:33.279: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:35.283: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:37.287: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:39.291: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:41.295: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:43.298: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:45.302: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:47.306: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:49.311: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:51.314: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:53.319: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:55.323: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:57.334: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:49:59.338: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:01.341: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:03.345: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:05.348: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:07.355: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:09.360: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:11.366: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:13.370: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:15.375: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:17.380: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:19.386: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:21.391: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:23.396: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:25.401: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:27.405: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:29.411: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:31.416: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:33.421: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:35.427: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:37.431: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:39.437: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:41.443: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:43.448: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:45.454: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:47.458: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:49.464: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:51.468: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:53.474: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:55.480: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:57.485: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:50:59.490: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:01.496: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:03.501: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:05.506: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:07.512: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:09.517: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:11.523: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:13.528: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:15.535: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:17.541: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:19.546: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:21.552: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:23.557: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:25.562: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
Apr  5 08:51:27.567: INFO: PersistentVolumeClaim rbd-pvc-new found but phase is Lost instead of Bound.
    Apr  5 08:55:20.115: failed to create new PVC: failed to wait for the pv and pvc to bind: PVC "rbd-pvc-new" did not become Bound: PersistentVolumeClaims [rbd-pvc-new] not all in phase Bound within 10m0s

```

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

nixpanic commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

nixpanic commented:
> /retest ci/centos/mini-e2e-helm/k8s-1.21

Again failed with #2327:
```
Apr  5 11:04:06.829: FAIL: backend images not matching kubernetes resource count,image count 22 kubernetes resource count 21
```

pkalever commented:
@Mergifyio refresh

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

pkalever commented:
@Mergifyio refresh

pkalever commented:
/restart ci/centos/mini-e2e-helm/k8s-1.22

pkalever commented:
/restart ci/centos/mini-e2e/k8s-1.21

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

pkalever commented:
/retest ci/centos/mini-e2e/k8s-1.21

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.23

pkalever commented:
/retest ci/centos/mini-e2e/k8s-1.21

pkalever commented:
/retest ci/centos/mini-e2e-helm/k8s-1.23

pkalever commented:
@Mergify refresh

pkalever commented:
@mergify rebase
